src/render.jl

```diff
@@ -47,12 +67,22 @@ function (colors::FlameColors)(s::Symbol)
 end
 
 function (colors::FlameColors)(nextidx, j::Integer, data)
-    colors.colorrt !== nothing && (data.status & runtime_dispatch) != 0 && return colors.colorrt
-    colors.colorgc !== nothing && (data.status & gc_event) != 0 && return colors.colorgc
-    colorvec = isodd(j) ? colors.colorsodd : colors.colorseven
     idx = nextidx[j]
-    nextidx[j] = mod1(idx+1, length(colorvec))
-    return colorvec[idx]
+    nextidx[j] = idx + 1
+    colorvec = nothing
+    if colors.colorrt !== nothing && (data.status & runtime_dispatch) != 0
+        colorvec = colors.colorrt
+    elseif colors.colorgc !== nothing && (data.status & gc_event) != 0
+        colorvec = colors.colorgc
+    end
+    if colorvec === nothing
+        colorvec = isodd(j) ? colors.colorsodd : colors.colorseven
+    else
+        isa(colorvec, Colorant) && return colorvec
+        n = (length(colorvec) + 1)÷2
+        idx = mod1(idx, n) + iseven(j) * n
```

Comment:
> If we're going to do this, should we just get rid of the `colorsodd` and `colorseven` fields and just have `colors`?

If we want the consistency or simple implementation, we should. However, I did this for compatibility. This commit breaks the definition of `FlameColors`, but we can use its default constructor without modifications. Of course, this "redundant" code can use all of the generated vectors.

Reply:
FlameGraphs is quite new. Now is definitely the time to make breaking changes, and field names are generally "internal" anyway. I'd say let's make that change, if you're up for doing it. (Or I can, but it will be a few days due to other commitments.)

Reply:
That makes sense. I'll change the `FlameColors`.